### PR TITLE
[RFC FS-1019] Fix 126 - Make "Module" suffix implicit if type being defined with the same name

### PR DESCRIPTION
**Pull request by dsyme**
This is the fix for 126, and also the implementation of RFC [FS-1019 - Make "Module" suffix implicit if type being defined with the same name](https://github.com/fsharp/FSharpLangDesign/blob/master/RFCs/FS-1019-implicitly-add-the-module-suffix.md)

This is a minor language addition but a useful one.

### CODE REVIEW

File: src/fsharp/TypeChecker.fs [hunk 1]
```diff
@@ -14154,7 +14163,7 @@ module EstablishTypeDefinitionCores =
 
         // Augmentations of type definitions are allowed within the same file as long as no new type representation or abbreviation is given 
         CheckForDuplicateConcreteType env id.idText id.idRange
-        CheckForDuplicateModule env id.idText id.idRange
+        //CheckForDuplicateModule env id.idText id.idRange
```
Comment (KevinRansom) [inline src/fsharp/TypeChecker.fs]: commented out code?

**Comment by dsyme** [comment]
Another transient failure in 
```
[00:39:28] 1) Failed : FSharp.Core.Unittests.FSharp_Core.Microsoft_FSharp_Control.AsyncModule.OnCancel.RaceBetweenCancellationHandlerAndDisposingHandlerRegistration
```

**Comment by Jand42** [comment]
This seems to rename the module even if the type with the same name is generic. (Tested with current local build of FCS)
It is not necessary in this case, and would be better to not have it renamed, so the name is the same from other languages.
I have not found a workaround (short of converting the module to a static class), both `CompiledName("OriginalName")` and `CompilationRepresentation(CompilationRepresentationFlags.None)` does not remove the suffix.

**Comment by dsyme** [comment]
@Jand42 Hmmm...  

- The interaction with ``CompiledName`` should indeed be at least mentioned in the RFC with the expected behavior made explicit.

- For generics: the RFC does say ``This applies whether the module is an abbreviation, or [the type is] with or without generic parameters.``  (I've added the clarification in square brackets).

**Pull request merged**